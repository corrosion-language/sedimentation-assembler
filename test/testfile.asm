section .bss
out: resb 10
section .text
global _start
_start:
	adc rdi, 0
	xor rdi, [out]
	mov cl, '0'
	lea rdi, [out]
	.L1:
		mov byte [rdi], cl
		inc rdi
		inc cl
		cmp cl, '9'
		jle .L1
	mov eax, 1
	mov edi, 1
	mov rsi, out
	mov edx, 10
	syscall

	xor ecx, ecx
	mov rdi, out
	.L2:
		add byte [rdi+rcx], 1
		inc rcx
		cmp rcx, 10
		jl .L2
	mov eax, 1
	mov edi, 1
	mov rsi, out
	mov edx, 10
	syscall

	xor ecx, ecx
	.L3:
		sub byte [ecx*2 + out + 1], 1
		inc ecx
		cmp ecx, 5
		jl .L3
	mov eax, 1
	mov edi, 1
	mov rsi, out
	mov edx, 10
	syscall

	mov eax, 60
	xor edi, edi
	syscall
