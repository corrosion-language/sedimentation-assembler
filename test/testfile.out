0123456789123456789:1133557799